WcaOnRails/spec/controllers/committees_controller_spec.rb

```diff
@@ -0,0 +1,221 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe CommitteesController, type: :controller do
+  let(:committee) { FactoryGirl.create :committee }
+  let(:committee_with_team) { FactoryGirl.create :committee, :with_team }
+
+  describe "GET #index" do
+    context "when not signed in" do
+      sign_out
+
+      it 'shows the committee index page' do
+        get :index
+        expect(response).to render_template :index
+      end
+    end
+
+    context "when signed in as admin" do
+      sign_in { FactoryGirl.create :admin }
+
+      it 'shows the committee index page' do
+        get :index
+        expect(response).to render_template :index
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+
+      it 'shows the committee index page' do
+        get :index
+        expect(response).to render_template :index
+      end
+    end
+  end
+
+  describe "GET #show" do
+    context "when not signed in" do
+      sign_out
+
+      it 'displays the committee show page' do
+        get :show, id: committee.slug
+        expect(response).to render_template :show
+      end
+    end
+
+    context "when signed in as admin" do
+      sign_in { FactoryGirl.create :admin }
+
+      it 'displays the committee show page' do
+        get :show, id: committee.slug
+        expect(response).to render_template :show
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+
+      it 'displays the committee show page' do
+        get :show, id: committee.slug
+        expect(response).to render_template :show
+      end
+    end
+  end
+
+  describe "GET #new" do
+    context "when not signed in" do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        get :new
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context "when signed in as admin" do
+      sign_in { FactoryGirl.create :admin }
+
+      it 'shows the committee new page' do
+        get :new
+        expect(response).to render_template :new
+      end
+    end
+
+    context 'when signed in as a regular user' do
+      sign_in { FactoryGirl.create :user }
+
+      it 'does not allow access' do
+        get :new
+        expect(response).to redirect_to root_url
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context "when not signed in" do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        post :create, committee: { name: "Software", email: "software@worldcubeassociation.org", duties: "Responsible for all WCA software" }
```

Comment:
See [this comment](https://github.com/cubing/worldcubeassociation.org/pull/725/files/#r77199556).